### PR TITLE
Refactor `JobCard` Props

### PR DESCRIPTION
**Pull request by alliu879**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Refactor `JobCard` props to include `filterList` callback function rather than `pendingJobs` and `filteredJobs` state hooks
- Rename `list` variables within `ApprovalScreen` and `FeedScreen` to `pendingJobs` and `approvedJobs` respectively

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
n/a

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
- Review `JobCard` first and then `ApprovalScreen` and `FeedScreen`
- Test by removing jobs in both `FeedScreen` and `ApprovalScreen`and ensuring that both are properly updated

## Next steps
n/a

## Relevant Links
n/a

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"
- Rationale for not passing in a generic function as a prop + how to pass in callback function as a prop: https://stackoverflow.com/a/36677798
- Documentation for `useCallback`: https://beta.reactjs.org/reference/react/useCallback

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
n/a

CC: @surajr10 @themiachou8

### CODE REVIEW

File: src/components/JobCard/JobCard.tsx [hunk 1]
```diff
@@ -12,20 +12,14 @@ interface JobCardProps {
   job: Job;
   idx: number;
   pending: boolean;
-  pendingJobs: Job[];
-  setPendingJobs: React.Dispatch<React.SetStateAction<Job[]>>;
-  filteredJobs: Job[];
-  setFilteredJobs: React.Dispatch<React.SetStateAction<Job[]>>;
+  filterList: (idx: number) => void;
```
Comment (wangannie) [inline src/components/JobCard/JobCard.tsx]: Nit: while you're at it with the cleanup, I'd also rename the `pending` prop to `isPending`

File: src/screens/Approval/ApprovalScreen.tsx [hunk 2]
```diff
@@ -33,22 +36,18 @@ function ApprovalScreen({
           alignItems: 'center',
           width: '100%'
         }}>
-        {list.map((job, index) => (
+        {pendingJobs.map((job, index) => (
             // eslint-disable-next-line react/jsx-key
```
Comment (wangannie) [inline src/screens/Approval/ApprovalScreen.tsx]: Please fix this!

File: src/screens/Feed/FeedScreen.tsx [hunk 3]
```diff
@@ -11,7 +11,7 @@ import { FeedStackScreenProps } from '../../types/navigation';
 
 function FeedScreen({ navigation }: FeedStackScreenProps<'FeedScreen'>) {
   const [open, setOpen] = useState(false);
-  const [list, setList] = useState([] as Job[]);
```
Comment (wangannie) [inline src/screens/Feed/FeedScreen.tsx:14]: nice

File: src/screens/Feed/FeedScreen.tsx [hunk 4]
```diff
@@ -11,7 +11,7 @@ import { FeedStackScreenProps } from '../../types/navigation';
...
+  const [approvedJobs, setApprovedJobs] = useState([] as Job[]);
   const [filteredList, setFilteredList] = useState([] as Job[]);
```
Comment (wangannie) [inline src/screens/Feed/FeedScreen.tsx]: I would also rename `filteredList` etc to indicate what specifically it's a list of e.g. `filteredApprovedJobs` if that's what it is (??). It's unclear what this is otherwise

File: src/screens/Approval/ApprovalScreen.tsx (hunk 2 above)
Comment (wangannie) [reply src/screens/Approval/ApprovalScreen.tsx]: It's still causing warnings in expo so you shouldn't avoid it

File: src/components/JobCard/JobCard.tsx [hunk 5]
```diff
@@ -40,7 +34,7 @@ function JobCard({
     } catch (e) {
       console.log(e);
     }
-    setPendingJobs(pendingJobs.filter((_, index) => index !== idx));
+    filterList(idx);
   }
 
   async function removeJob() {
```
Comment (wangannie) [inline src/components/JobCard/JobCard.tsx:40]: (unrelated to jobcard props, more a general question) why are we removing the job from firebase from the FeedScreen? What is the use case there? cc @surajr10

Comment (wangannie) [inline src/components/JobCard/JobCard.tsx:40]: If I'm understanding correctly, this is your current design:

We have a callback in the parent (ApprovalScreen/FeedScreen) to remove the card from the list. In the child (JobCard), we hide the modal, trigger the remove firebase query, and then call that callback function to remove the card from the list.

I think splitting the 2 remove actions (query + remove from list) is really unintuitive, so I don't think we should split that logic -- instead, I would just lift all the removal-related logic into the callback from the parent. Then when you click **REMOVE**, it'd just call `setModalVisible(false)` and trigger your `onRemove` callback which contains all the removal logic.

File: src/components/JobCard/JobCard.tsx (hunk 1 above)
Comment (wangannie) [inline src/components/JobCard/JobCard.tsx]: Since this is a callback, I'd recommend falling the `on{ACTION}` convention by naming this `onRemove`.

Calling it `filterList` makes it seem like there's a list contained in `JobCard` and that function is used to filter it.

File: src/screens/Approval/ApprovalScreen.tsx [hunk 6]
```diff
@@ -10,18 +10,21 @@ import { ApprovalStackScreenProps } from '../../types/navigation';
...
   useEffect(() => {
     const fetchJobs = async () => {
       const data = await getAllJobs('notApprovedJobs');
-      setList(data);
+      setPendingJobs(data);
     };
     fetchJobs();
   }, [isFocused]);
 
+  const filterPendingJobs = useCallback((idx: number) => {
```
Comment (wangannie) [inline src/screens/Approval/ApprovalScreen.tsx]: Rename to `handleRemovePendingJob` or `handleRemoveJob`

File: src/screens/Feed/FeedScreen.tsx [hunk 7]
```diff
@@ -34,19 +34,23 @@ function FeedScreen({ navigation }: FeedStackScreenProps<'FeedScreen'>) {
...
-      setFilteredList(list);
+      setFilteredList(approvedJobs);
     } else {
-      setFilteredList(list.filter(job => job.category === category));
+      setFilteredList(approvedJobs.filter(job => job.category === category));
     }
-  }, [category, list]);
+  }, [category, approvedJobs]);
+
+  const filterApprovedJobs = useCallback((idx: number) => {
```
Comment (wangannie) [inline src/screens/Feed/FeedScreen.tsx:51]: Rename to `handleRemoveApprovedJob` or `handleRemoveJob`

File: src/screens/Feed/FeedScreen.tsx [hunk 8]
```diff
@@ -76,12 +80,9 @@ function FeedScreen({ navigation }: FeedStackScreenProps<'FeedScreen'>) {
 
         {filteredList.map((job, index) => (
           // eslint-disable-next-line react/jsx-key
```
Comment (wangannie) [inline src/screens/Feed/FeedScreen.tsx]: Fix this!

**Review by wangannie — COMMENTED** [review]
It's looking good! I noted one bigger restructuring suggestion + a couple of nits on naming clarity. Let me know if you want to discuss anything further.

Since your PR doesn't touch any packages, you should also revert all changes to `yarn.lock` in this PR.

File: src/components/JobCard/JobCard.tsx (hunk 5 above)
Comment (alliu879) [reply src/components/JobCard/JobCard.tsx:40]: I found that it was necessary to split the logic as such in order to use `filteredApprovedJobs`/`notApprovedJobs` state hooks from within `JobCard` without passing the hooks in directly. Is there a different way this can be done?

Comment (alliu879) [reply src/components/JobCard/JobCard.tsx:40]: Not sure if I completely understand your question? We remove the job from firebase in `JobCard`, not `FeedScreen`; we want to allow admins to do this to better manage the feed.
